p2p/src/net/libp2p/backend.rs

```diff
@@ -212,7 +256,7 @@ impl Backend {
                 match self.swarm.disconnect_peer_id(peer_id) {
                     Ok(_) => {
                         log::trace!("peer {:?} disconnected", peer_id);
-                        self.established_conns.remove(&peer_id);
+                        self.swarm.behaviour_mut().established_conns.remove(&peer_id);
```

Comment:
here too...

I think this fiasco was started by separating objects but not separating behavior.